app/models/item.js

```diff
@@ -0,0 +1,30 @@
+var mongoose = require('mongoose');
+var Schema = mongoose.Schema;
+var Types = Schema.Types;
+var ObjectId = Types.ObjectId;
+
+var QueryPlugin = require('mongoose-query');
+
+var ItemSchema = new Schema({
+  barcode : { type: String, required:true, unique:true },
+  name : { type: String, required:true },
```

Comment:
name can and should be unique. e.g. if it is cable we should describe name as much as it become unique like "USB - miniUSB -  15cm"...